### ISSUE TITLE
Object#load Argument Error

### ISSUE THREAD

**Issue by waynerobinson** [metaskills/store_configurable#2]
There seems to be an issue with Object#load. I am getting the following error when creating new objects, etc.

``````
ArgumentError: wrong number of arguments (1 for 2)
    from /Users/waynerobinson/.rvm/gems/ruby-1.9.3-p125/gems/store_configurable-3.2.1/lib/store_configurable/object.rb:30:in `load'
    from /Users/waynerobinson/.rvm/gems/ruby-1.9.3-p125/gems/activerecord-3.2.2/lib/active_record/attribute_methods/serialization.rb:24:in `unserialize'
    from /Users/waynerobinson/.rvm/gems/ruby-1.9.3-p125/gems/activerecord-3.2.2/lib/active_record/attribute_methods/serialization.rb:15:in `unserialized_value'
    from /Users/waynerobinson/.rvm/gems/ruby-1.9.3-p125/gems/activerecord-3.2.2/lib/active_record/attribute_methods/read.rb:82:in `__temp__'
    from /Users/waynerobinson/.rvm/gems/ruby-1.9.3-p125/gems/activerecord-3.2.2/lib/active_record/attribute_methods/read.rb:46:in `type_cast_attribute'
    from /Users/waynerobinson/.rvm/gems/ruby-1.9.3-p125/gems/activerecord-3.2.2/lib/active_record/attribute_methods/read.rb:126:in `read_attribute'
    from /Users/waynerobinson/.rvm/gems/ruby-1.9.3-p125/gems/activerecord-3.2.2/lib/active_record/attribute_methods.rb:200:in `attribute_for_inspect'
    from /Users/waynerobinson/.rvm/gems/ruby-1.9.3-p125/gems/activerecord-3.2.2/lib/active_record/base.rb:640:in `block in inspect'
    from /Users/waynerobinson/.rvm/gems/ruby-1.9.3-p125/gems/activerecord-3.2.2/lib/active_record/base.rb:638:in `collect'
    from /Users/waynerobinson/.rvm/gems/ruby-1.9.3-p125/gems/activerecord-3.2.2/lib/active_record/base.rb:638:in `inspect'
    from /Users/waynerobinson/.rvm/gems/ruby-1.9.3-p125/gems/wirble-0.1.3/lib/wirble.rb:432:in `output_value'
    from /Users/waynerobinson/.rvm/gems/ruby-1.9.3-p125/gems/railties-3.2.2/lib/rails/commands/console.rb:47:in `start'
    from /Users/waynerobinson/.rvm/gems/ruby-1.9.3-p125/gems/railties-3.2.2/lib/rails/commands/console.rb:8:in `start'
    from /Users/waynerobinson/.rvm/gems/ruby-1.9.3-p125/gems/railties-3.2.2/lib/rails/commands.rb:41:in `<top (required)>'
    from script/rails:6:in `require'
    from script/rails:6:in `<main>'```

It appears that ActiveRecord's Serialization objects call load with only one parameter and do not include the 'owner' parameter.
``````

**Comment by metaskills** [comment]
Odd, do the test not work for you? Or do the test run and the actual usage is blowing up?

**Comment by metaskills** [comment]
I see you are using the console. Can you show me what you did to generate that stack trace?

BTW, I know about ActiveRecord serialization load taking only one arg. This should not be a problem because our `config` read method which delegates to the `_config` override will perform a singleton replacement of load. See here.

https://github.com/metaskills/store_configurable/blob/master/lib/store_configurable/read.rb#L25-51

That said, I am curious as to how you generated this error.

**Comment by waynerobinson** [comment]
All I am doing to produce this error is calling "ModelName.new" where the model has store_configurable in it.

**Comment by metaskills** [comment]
Well how about that :/

``` diff
diff --git a/test/cases/base_test.rb b/test/cases/base_test.rb
index 0c9d88c..5fb8f7f 100644
--- a/test/cases/base_test.rb
+++ b/test/cases/base_test.rb
@@ -6,6 +6,10 @@ class StoreConfigurable::BaseTest < StoreConfigurable::TestCase
     new_user.config.wont_be_nil
   end

+  it 'can inspect a new user' do
+    new_user.inspect
+  end
+  
```

Now that I have a failing test, let me fix it.

**Comment by metaskills** [comment]
OK, published v3.2.2 which fixes that.

**Comment by waynerobinson** [comment]
Wonderful. Thanks for this. I have implemented bits and pieces of a library like this for specific use cases in the past but this recursive version is much more useful.
